ocs/ocp.py

```diff
@@ -134,23 +148,48 @@ def apply(self, yaml_file):
         command = f"apply -f {yaml_file}"
         return self.exec_oc_cmd(command)
 
-    def wait_for_resource_status(
-        self, resource_name, condition, timeout=30, sleep=3
+    def wait(
+        self, condition, resource_name='', selector=None, resource_count=0,
+        timeout=60, sleep=3
     ):
         """
         Wait for a resource to reach to a desired condition
 
         Args:
-            resource_name (str): The name of the resource to wait
-                for (e.g.my-pv1)
             condition (str): The desired state the resource should be at
                 This is referring to: status.phase presented in the resource
                 yaml file
                 (e.g. status.phase == Available)
+            resource_name (str): The name of the resource to wait
+                for (e.g.my-pv1)
+            selector (str): The resource selector to search with.
+                Example: 'app=rook-ceph-mds'
+            resource_count (int): How many resources expected to be
+
+        Returns:
+            bool: True in case all resources reached desired condition,
+                False otherwise
+
         """
         for sample in TimeoutSampler(
-            timeout, sleep, self.get, resource_name
+            timeout, sleep, self.get, resource_name, True, selector
         ):
-            if sample.status.phase == condition:
-                return True
+            # More than 1 resources returned
+            if sample.kind == 'List':
+                in_condition = []
+                sample = sample['items']
+                for item in sample:
+                    if item.status.phase == condition:
+                        in_condition.append(item)
+                    if resource_count:
+                        if len(in_condition) == resource_count and (
+                            len(sample) == len(in_condition)
+                        ):
+                            return True
+                    elif len(sample) == len(in_condition):
+                        return True
+            # Only 1 resource returned
+            else:
```

Comment:
what if you had resource_count set to 2? I think this case you should return True in this else branch and continue to sampling

Reply:
Not sure I understand

Reply:
You have parameter resource_count right? I understand it that you can wait til 2 resources are created.
So you wait for 2 resources for example. But on line [178](https://github.com/red-hat-storage/ocs-ci/pull/70/files#diff-9d7ae45bb7efabea0529c53dbca720f9R178) you are checking if kind is List.

I am not sure what is the behavior here, but if it won't be list and only one resource returned as you mentioned on line 191 you are going directly to else block.
checking it's status and if it's == condition returning `True`.

So you actually ending your wait with True status, but you have only 1 resource created but expected to wait for 2 resources.

Do I missing something or misunderstand this function?

Reply:
No you are right.
Fixing now and nice catch!

Reply:
So it seems like that List is always returned from this command:
2019-05-22 16:25:05,073 - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /tmp/ratamir-test-ocs4-31174/auth/kubeconfig get Pod --selector=app=rook-ceph-mdsa -o yaml
DEBUG:utility.utils:CMD output: apiVersion: v1
items: []
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""

Reply:
Ok found the thing:
1) when providing resource_name you are expected to get 1 resource 
2) when using selector --> can have multiple so List is returned